pkg/common/rotationutil/rotationutil.go

```diff
@@ -22,37 +28,46 @@ func X509Expired(now time.Time, cert *x509.Certificate) bool {
 // JWTSVIDExpiresSoon determines if the given JWT SVID should be rotated
 // based on presented current time, the JWT's expiration.
 // Also returns true if the JWT is already expired.
-func JWTSVIDExpiresSoon(svid *client.JWTSVID, now time.Time) bool {
+func JWTSVIDExpiresSoon(svid *client.JWTSVID, now time.Time, availabilityTarget time.Duration, log logrus.FieldLogger) bool {
 	if JWTSVIDExpired(svid, now) {
 		return true
 	}
 
-	// if the SVID has less than half of its lifetime left, consider it
-	// as expiring soon
-	return shouldRotate(now, svid.IssuedAt, svid.ExpiresAt)
+	// if the SVID has less than half of its lifetime left or reaches the availability target,
+	// consider it as expiring soon
+	return shouldRotate(now, svid.IssuedAt, svid.ExpiresAt, availabilityTarget, log)
 }
 
 // JWTSVIDExpired returns true if the given SVID is expired.
 func JWTSVIDExpired(svid *client.JWTSVID, now time.Time) bool {
 	return !now.Before(svid.ExpiresAt)
 }
 
-func shouldRotate(now, beginTime, expiryTime time.Time) bool {
+func shouldRotate(now, beginTime, expiryTime time.Time, availabilityTarget time.Duration, log logrus.FieldLogger) bool {
 	ttl := expiryTime.Sub(now)
 	// return true quickly if the expiry is already met.
 	if ttl <= 0 {
 		return true
 	}
 
-	halfLife := halfLife(beginTime, expiryTime)
+	if shouldFallbackDefaultRotationStrategy(beginTime, expiryTime, availabilityTarget) {
+		// calculate a jitter delta to spread out rotations
+		halfLife := halfLife(beginTime, expiryTime)
+		delta := jitterDelta(halfLife)
+		min := halfLife - delta
+		jitteredHalfLife := time.Duration(rand.Int63n(int64(delta)*2) + int64(min)) //nolint // gosec: no need for cryptographic randomness here
 
-	// calculate a jitter delta to spread out rotations
-	delta := jitterDelta(halfLife)
-	min := halfLife - delta
+		if ttl <= jitteredHalfLife {
+			if availabilityTarget > 0 {
+				log.Warn("SVID lifetime isn't long enough to guarantee the availability_target, fell back to the default rotation strategy")
+			}
+			return true
+		}
 
-	jitteredHalfLife := time.Duration(rand.Int63n(int64(delta)*2) + int64(min)) //nolint // gosec: no need for cryptographic randomness here
+		return false
+	}
 
-	return ttl <= jitteredHalfLife
+	return ttl <= availabilityTarget
```

Comment:
What do you think, should we introduce the jitter for availability target ?
IMO: 
- ±10% jitter is too large and may shift the availability target.

Reply:
yes, I think this may have a jitter, this can even simplify your code here too, because you can get halftime or your `availabilityTarget` as initial step for the shitter.
if you are worried about 10%, you can reduce that amount when availabilityTarget is set,
what value do you think is good enough?

Reply:
I also think that we should preserve the jitter (with a reduced percentage in this case).

Reply:
Since availability_target is more than 24 hours, what about the using absolute maximum jitter instead of percentage?  For example, 0 ~ +10min.

https://github.com/spiffe/spire/pull/4599/files#diff-cfe3084f16541101db16f11e483a339dcfecd43d30e7389b7e9ddb80537e8cccR114